src/Language/PureScript/Make/Actions.hs

```diff
@@ -269,7 +271,7 @@ buildMakeActions outputDir filePathMap foreigns usePrefix =
   requiresForeign = not . null . CF.moduleForeign
 
   progress :: ProgressMessage -> Make ()
-  progress = liftIO . putStrLn . renderProgressMessage
+  progress = liftIO . TIO.hPutStrLn stderr . renderProgressMessage
```

Comment:
This change is what's causing the problem. I'm not entirely sure why yet.

Reply:
I think it's because the `hPutStrLn` in `Data.Text.IO` is implemented like this:

```haskell
hPutStrLn h t = hPutStr h t >> hPutChar h '\n'
```

See https://hackage.haskell.org/package/text-1.2.4.0/docs/src/Data.Text.IO.html#hPutStrLn

and the runtime can switch to a different thread after the `hPutStr` but before the `hPutChar`. Perhaps `liftIO . TIO.hPutStr stderr . (<> "\n) . renderProgressMessage` might work?

Reply:
Yeah, that fixes it.